src/praveen/vehicleDirection.test.ts

```diff
@@ -0,0 +1,51 @@
+import { getFinalDirectionForVehicle, getNextDirectionForMove } from './vehicleDirection';
+
+const testData = [
+  'STRAIGHT',
+  'RIGHT',
+  'LEFT',
+  'BACK',
+  'RIGHT',
+  'STRAIGHT',
+  'LEFT'
+];
+
+describe('testing getFinalDirectionForVehicle function', function () {
+  it('testing if testData returns SOUTH direction', function () {
+    const result = getFinalDirectionForVehicle(testData);
+    expect(result).toEqual('SOUTH');
+  });
+
+  it('testing if it returns the NORTH direction with no moves', function () {
+    const result = getFinalDirectionForVehicle([]);
+    expect(result).toEqual('NORTH');
+  });
+
+  it('testing if it returns exception for invalid moves', function () {
+    try {
+      getFinalDirectionForVehicle(['TOP']);
+    } catch (e) {
+      expect(e.message).toBe('Invalid Move');
```

Comment:
If I am right, this only checks the error thrown has the right error message or not but not whether error is thrown. If the function don't throw any error this test case still passes.
We have a reference which do the similar thing at [PR#10](https://github.com/unifystockx/code_compete/pull/10/files#diff-f927f7641dd8233b5399624b20e84473) : Line 32

Reply:
fixed